src/main/integration/java/com.juniorstart.juniorstart/Service/UserProfileTest.java

```diff
@@ -0,0 +1,127 @@
+package com.juniorstart.juniorstart.service;
+
+import com.juniorstart.juniorstart.exception.BadRequestException;
+import com.juniorstart.juniorstart.model.*;
+import com.juniorstart.juniorstart.repository.UserDao;
+import com.juniorstart.juniorstart.repository.UserProfileRepository;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.test.context.junit4.SpringRunner;
+import java.util.*;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+/** Represents an user service.
+ * @author Rafał Maduzia
+ * @version 1.0
+ */
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@AutoConfigureTestDatabase(replace= AutoConfigureTestDatabase.Replace.NONE)
+public class UserProfileTest {
+
+    @SpyBean
+    private UserProfileService userProfileService;
+
+    @Autowired
+    private UserDao userRepository;
+
+    @Autowired
+    private UserProfileRepository userProfileRepository;
+
+    List<String> technologyList  = new ArrayList<>();
```

Comment:
listę powinno się inicjować w metodzie setUp

Reply:
A w tym przypadku to dobry pomysł? Bo do listy dopisuję różne wartości. Gdybym przypisał je w metodzie setUp, to później musiałbym się odwoływać po indeksach, gdybym chciał wyciągnąć jedną spośród innych.

Reply:
Raczej dobry pomosł.

Reply:
@hosu794 
W tym przypadku, to chyba jednak lipa, bo się robi np taki kaszalot:
          (new ArrayList<String>(Collections.singletonList(technologyList.get(0))),
                        new ArrayList<UserRole>(Collections.singletonList(userRoleListEnum.get(0)))))

Czy to tylko mnie razi w oczy ? Co prawda mogę dopisać gdzieś linię w komentarzu co jest co